Microsoft.Research/Contracts/MsCorlib/System.IO.Directory.cs

```diff
@@ -91,7 +91,21 @@ public static String[] GetLogicalDrives()
     }
 #endif
 
-#if !SILVERLIGHT_4_0 && !SILVERLIGHT_5_0
+#if !SILVERLIGHT
+    public static String[] GetFileSystemEntries(string path, string searchPattern, SearchOption searchOption)
+    {
+      Contract.Requires(!String.IsNullOrEmpty(path));
+      Contract.Requires(searchPattern != null);
+      Contract.Requires(searchOption == SearchOption.AllDirectories || searchOption == SearchOption.TopDirectoryOnly);
+      Contract.Ensures(Contract.Result<string[]>() != null);
+      Contract.Ensures(Contract.ForAll(Contract.Result<string[]>(), file => file != null));
+      Contract.EnsuresOnThrow<System.IO.IOException>(true, @"path is a file name.");
```

Comment:
Is this message is correct? It seems slightly misleading to me because it doesn't tell what the error type is...